### PR TITLE
Add Azure OpenAI Service support: specific deployment id by model.

### PR DESCRIPTION
**Pull request by suensummit**
### Background

Since each deployment is created according to one specific model in Azure OpanAI Service, the deployment id should be specific by model.

### Changes

In `.env.template`, modify `OPENAI_DEPLOYMENT_ID` to `FAST_LLM_MODEL_DEPLOYMENT_ID` and `SMART_LLM_MODEL_DEPLOYMENT_ID`.
Add a function `get_openai_deployment_id` in `llm_utils.py` to specific deployment id by model (fast/smart).
Also updated README.

### Change Safety

- [ ] I have added tests to cover my changes
- [x] I have considered potential risks and mitigations for my changes

It's more like a modification to fit API requirement, does not affect core functionality.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (nponeccop) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: This doesn't look right. This is a general file, not specific to this PR

**Review by nponeccop — CHANGES_REQUESTED** [review]
Remove the PULL_REQUEST_TEMPLATE changes from the PR.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (suensummit) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: OK thanks

**Review by nponeccop — APPROVED** [review]

**Comment by nponeccop** [comment]
There are conflicts again.

File: .gitignore
Comment (nponeccop) [inline .gitignore:1]: An unrelated change creeped in. Roll it back.

File: scripts/ai_functions.py
Comment (nponeccop) [inline scripts/ai_functions.py:1]: Unrelated changes again. They make it harder to review.

File: scripts/main.py
Comment (nponeccop) [inline scripts/main.py:1]: Another unrelated change

File: scripts/speak.py [hunk 1]
```diff
@@ -7,7 +7,6 @@
 import threading
 from threading import Lock, Semaphore
 
-
```
Comment (nponeccop) [inline scripts/speak.py:10]: One more.

**Comment by suensummit** [comment]
> There are conflicts again.

yeah they are all come from rebase conflicts and it will be quite frustrated if @Torantulino does not intent to merge this PR

**Pull request closed without merging**